access-modifier-checker/src/main/java/org/kohsuke/accmod/impl/Checker.java

```diff
@@ -138,85 +138,22 @@ public void loadRestrictions(InputStream stream, final boolean isInTheInspectedM
         BufferedReader r = new BufferedReader(new InputStreamReader(stream, "UTF-8"));
         String className;
         while ((className=r.readLine())!=null) {
-            InputStream is = dependencies.getResourceAsStream(className.replace('.','/') + ".class");
-            if (is==null) {
-                errorListener.onWarning(null,null,"Failed to find class file for "+ className);
-                continue;
-            }
-
-            try {
-                new ClassReader(is).accept(new ClassVisitor(Opcodes.ASM5) {
```

Comment:
Can you please revert what seems to be an unrelated extract refactoring which makes this very hard to review?

Reply:
not unnecessary, I had to extract a superclass to share code between the class and the package based visitor

Reply:
to give some details, RestrictionsClassVisitor + AnnotatedClassVisitor is == to the initial anonymous class, but this parent class allows to share code with AnnotatedPackageVisitor, which is the sole new code here